Client/Application/WebServer.swift

```diff
@@ -18,8 +18,10 @@ class WebServer {
     }
 
     func start() -> Bool {
-        if !server.running {
-            try! server.startWithOptions([GCDWebServerOption_Port: 6571, GCDWebServerOption_BindToLocalhost: true, GCDWebServerOption_AutomaticallySuspendInBackground: true])
+        withExtendedLifetime(server) {
```

Comment:
I'm not entirely sure either, but it worked. It seems to ensure that the GCDWebServer stuck around long enough that it was still present when `beginBackgroundTaskWithExpirationHandler` is executed later. 

It was a little bit of a fix in the dark - the proper solution was to ensure that the `self` used in teh `beginBackgroundTaskWithExpirationHandler` block was a strong reference using the  `__weak typeof(self) weakself = self;...` pattern, but I didn't want to fork GCDWebServer for that purpose. The function that calls the backgrounding method is executed after listening for the ApplicationDidEnterForeground notification and is therefore not kicked off by us, so there was no way of ensuring in our code that the webserver was always present.
As the notification was set during start, forcing the server to stick around seemed like something to try, and it worked.
Happy to see if I can find some other way of doing this.